### PR TITLE
Hosted Gallery more fixes

### PR DESCRIPTION
**Pull request by lps88**
## What does this change?
- add tracking when user moves between images
- add gradient behind gallery title & cta
- twitter share button should now work (error due to unescaped #)
- move call to action markup into a separate template as we use it twice
- better text sizes, and new footer styling for mobile breakpoint
  before:
  image
  after:
  image
- info button turns into close button when caption is open on mobile
  before:
  image
  after:
  image
- new styling for up/down arrows
  before:
  image
  after:
  image
## Request for comment

@Calanthe

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js [hunk 1]
```diff
@@ -43,7 +45,7 @@ define([
         // ELEMENT BINDINGS
         this.$galleryEl = $('.js-hosted-gallery-container');
         this.$imagesContainer = $('.js-hosted-gallery-images', this.$galleryEl);
-        this.$captionContainer = $('.js-hosted-gallery-captions');
+        this.$captionContainer = $('.hosted-gallery__lower-bar');
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js]: Why did you remove the `js-` part?

File: commercial/app/views/hosted/guardianHostedGallery.scala.html [hunk 2]
```diff
@@ -10,34 +11,27 @@
                 <div class="js-hosted-gallery-container hosted-gallery__gallery-container">
                     <div class="js-hosted-gallery-images hosted-gallery__images-container">
                         @guardianHostedGalleryImage(page.images.head, {
+                            <div class="hosted-gallery__image-gradient"></div>
```
Comment (regiskuckaertz) [inline commercial/app/views/hosted/guardianHostedGallery.scala.html]: Do you really need an extra DIV for this? Elements can have multiple background images in CSS.

File: commercial/app/views/hosted/guardianHostedGalleryCta.scala.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+@(ctaText: String, ctaLink: String, dataLinkName: String, cssClass: String = "")
+    <div class="hosted-gallery__cta @{cssClass}">
+        <div class="hosted-gallery__cta-title">
+            @{ctaText}
+        </div>
+        <a class="hosted-gallery__cta-link" href="@{ctaLink}" target="_blank" data-link-name="@{dataLinkName}">
+            Visit OMGB now @fragments.inlineSvg("arrow-right", "icon")
```
Comment (regiskuckaertz) [inline commercial/app/views/hosted/guardianHostedGalleryCta.scala.html]: Shouldn't this text be abstracted too?

File: commercial/app/views/hosted/guardianHostedGallery.scala.html (hunk 2 above)
Comment (lps88) [inline commercial/app/views/hosted/guardianHostedGallery.scala.html]: I hadn't thought about that, I'll update it 😺

File: commercial/app/views/hosted/guardianHostedGalleryCta.scala.html (hunk 3 above)
Comment (lps88) [inline commercial/app/views/hosted/guardianHostedGalleryCta.scala.html]: Yes........ oops

File: static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js [hunk 4]
```diff
@@ -275,7 +277,7 @@ define([
                 if(this.useSwipe){
                     this.translateContent(this.index, 0, 100);
                 }
-
+                omniture.trackLinkImmediate(config.page.analyticsName + ' - image ' + this.index);
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js]: can you use a `data-link-name` attribute instead? that still ends up calling `trackLink`, but normalises the name

File: commercial/app/views/hosted/guardianHostedGallery.scala.html (hunk 2 above)
Comment (lps88) [inline commercial/app/views/hosted/guardianHostedGallery.scala.html]: Ah I think it's because it needed to be on top of the `img` tag but underneath the text, so didn't really fit anywhere else

Comment (regiskuckaertz) [inline commercial/app/views/hosted/guardianHostedGallery.scala.html]: Hmm, I see. Can you add a class to `hosted-gallery__cta` that adds the gradient?

File: static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js (hunk 4 above)
Comment (lps88) [inline static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js]: This is called when changing between images (can be by scrolling, swiping or arrow buttons) which is all controlled by the javascript. So, nowhere to really put that attribute. Maybe trackLinkImmediate isn't the right omniture method to call? I'm not familiar at all with omniture! What do you think is best?

File: commercial/app/views/hosted/guardianHostedGallery.scala.html (hunk 2 above)
Comment (lps88) [inline commercial/app/views/hosted/guardianHostedGallery.scala.html]: Yeah I think I've found a way round it 🐻

**Comment by Calanthe** [comment]
👍
